docs/docs/daft.md

```diff
@@ -0,0 +1,148 @@
+
+For Python users, Daft is complementary to PyIceberg as a query engine layer:
+
+* **PyIceberg:** catalog/table management tasks (e.g. creation of tables, modifying table schemas)
+* **Daft:** querying tables (e.g. previewing tables, data ETL and analysis)
+
+In database terms, PyIceberg is the Data Description Language (DDL) for database administration and Daft is the Data Manipulation Language (DML) for querying data.
+
+## Enabling Iceberg support in Daft
+
+To use Iceberg with Daft, simply ensure that the [PyIceberg](https://py.iceberg.apache.org/) library is also installed in your current Python environment.
+
+```
+pip install getdaft pyiceberg
+```
+
+## Querying Iceberg using Daft
+
+### Reading PyIceberg tables
+
+Daft interacts natively with [PyIceberg](https://py.iceberg.apache.org/) to read from Iceberg.
+
+Simply load a PyIceberg table and pass it into Daft as follows:
+
+``` py
+import daft
+from pyiceberg import load_catalog
+
+table = load_catalog("my_catalog").load_table("my_tpch_namespace.lineitem")
+df = daft.read_iceberg(table)
+df = df.select("L_SHIPDATE", "L_ORDERKEY", "L_COMMENT")
+df.show()
+```
+
+```
+╭────────────┬────────────┬────────────────────────────────╮
+│ L_SHIPDATE ┆ L_ORDERKEY ┆ L_COMMENT                      │
+│ ---        ┆ ---        ┆ ---                            │
+│ Date       ┆ Int64      ┆ Utf8                           │
+╞════════════╪════════════╪════════════════════════════════╡
+│ 1992-01-02 ┆ 2186280097 ┆ ions sleep about the si        │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1992-01-02 ┆ 175366628  ┆ gular accoun                   │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1992-01-02 ┆ 2186602151 ┆ blithely even                  │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1992-01-02 ┆ 3937663654 ┆ ake boldly among the ideas. s… │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1992-01-02 ┆ 2186781220 ┆ thely. slyly pending ideas ar… │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1992-01-02 ┆ 3937999493 ┆  haggle at the regular, pen    │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1992-01-02 ┆ 2186933061 ┆ ickly. slyly                   │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1992-01-02 ┆ 3938167204 ┆ carefully silent instructions… │
+╰────────────┴────────────┴────────────────────────────────╯
+
+(Showing first 8 rows)
+```
+
+Any subsequent filter operations on the Daft `df` DataFrame object will be correctly optimized to take advantage of Iceberg features such as hidden partitioning and file-level statistics for efficient reads.
+
+``` py
+import datetime
+
+# Filter which takes advantage of partition pruning capabilities of Iceberg
+df = df.where(df["L_SHIPDATE"] > datetime.date(1993, 1, 1))
+df.show()
```

Comment:
Try to update this with a date field...I was trying to materialize the snapshot table and I had just found the from_pylist method but not sure if it will work with a custom object since it is a List of Dictionaries.

Reply:
We currently only support lists of dictionaries for `from_pylist`. I'll figure out the field with the Spark demo!